packages/types/src/type/ExtrinsicEra.ts

```diff
@@ -2,33 +2,156 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+import EnumType from '../codec/EnumType';
+import {isHex, isU8a, hexToU8a, isObject} from '@polkadot/util';
+import Tuple from '../codec/Tuple';
+import U64 from '../primitive/U64';
+import U8 from '../primitive/U8';
+import U16 from '../primitive/U16';
 import { AnyU8a } from '../types';
+import BlockNumber from "@polkadot/types/type/BlockNumber";
+import U8a from "@polkadot/types/codec/U8a";
 
-import { u8aToU8a } from '@polkadot/util';
+interface EraMethod {
+  startBlockNumber: BlockNumber;
+  endBlockNumber: BlockNumber;
+}
+
+export default class ExtrinsicEra extends EnumType<ImmortalEra | MortalEra> {
+  constructor (value?: any, index?: number) {
+    super({ ImmortalEra, MortalEra }, value, index);
+  }
+
+  /**
+   * @description Returns the item as a [[ImmortalEra]]
+   */
+  get asImmortalEra (): ImmortalEra | undefined{
+    if (this.index === 0) {
+      return this.value as ImmortalEra;
+    }
+    return;
+  }
+
+  /**
+   * @description Returns the item as a [[MortalEra]]
+   */
+  get asMortalEra (): MortalEra | undefined {
+    if (this.index === 1) {
```

Comment:
check other class which extends EnumTypes, i think they use asset() which throws error if index not match.

Reply:
done